Sources/Classes/RSDeviceModeManager.m

```diff
@@ -21,11 +21,20 @@ - (instancetype)initWithConfig:(RSConfig *) config andDBPersistentManager:(RSDBP
         self->dbPersistentManager = dbPersistentManager;
         self->networkManager = networkManager;
         self->integrationOperationMap = [[NSMutableDictionary alloc] init];
-        self->eventReplayMessage = [[NSMutableDictionary alloc] init];
+        self->beforeSDKInitEventTimestamp = [[RSPreferenceManager getInstance] getBeforeSDKInitEventTimestamp];
+        if (self->beforeSDKInitEventTimestamp == 0) {
+            self->beforeSDKInitEventTimestamp = [RSUtils getTimeStampLong];
+            [[RSPreferenceManager getInstance] saveBeforeSDKInitEventTimestamp:self->beforeSDKInitEventTimestamp];
```

Comment:
Do we think `RSDeviceModeManager.m`  is the right place for  `beforeSDKInitEventTimestamp`/`previousSDKInitTimeStamp`? (I suggest moving to EventRepository or RSPrefereceManager itself similar to `getAnonymousId` implementation)

Reply:
IMO, it is indeed the right place, as the significance of this `beforeSDKInitEventTimestamp` value comes into effect only when any events are made before SDK init, and as events are processed in the DeviceModeManager class, it should be responsible to maintain this property.

Anyway, the logic would be intact, even if we place it in the EventRepository class. If you think it would be more appropriate to place this variable in the `EventRepository` class do LMK!

Reply:
Based on the discussion, we are going to completely get rid of this timestamp logic and we will basically refactor the past unprocessed event deletion logic in the EventRepository init method.